### DISCUSSION TITLE
Data retention limit (self-hosted)

### DISCUSSION THREAD

**Discussion by Cellaryllis** [plausible/analytics#1940] [Anything Else]
Hello!

I haven't been able to find so in the current implementation, but is there a way to run the server while automatically deleting entries/data older than x days? We're only interested in recent data (30 days) and we'd rather not needlessly grow the database out of proportion.

If there's no way, and currently no plan on adding this, what would be the steps to setup a cron job to automatically delete old entries?
Since this is using postgresql and clickhouse, I assume I could run a cron job that executes a query that wipes the data? But I'm wholly unfamiliar with postgresq and clickhouse and the data structure of the logged stats/data and any guidance/pointers would be appreciated!

**Comment by noelforte** [comment]
Stumbled upon this thread with the same issue as @Cellaryllis, I'm getting into self-hosting and am also keen to put some limits on the size of my database either via a hardcoded retention policy option in the app or via a cron job that will regularly clear out old entries. This seems possibly related to #2147, so interested to hear from the maintainers whether that feature could help with this.
